boardgamewe-ui/src/Login/LoginLayout.js

```diff
@@ -173,6 +187,64 @@ class LoginLayout extends React.Component {
 
     };
 
+    validateField(fieldName, value) {
+        let fieldValidationErrors = this.state.formErrors;
+        let emailValid = this.state.emailValid;
+        let passwordValid = this.state.passwordValid;
+        let usernameValid = this.state.usernameValid;
+        let surnameValid = this.state.surnameValid;
+        let nameValid = this.state.nameValid;
+
+        switch(fieldName) {
+            case 'email':
+                emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
+                fieldValidationErrors.email = emailValid ? '' : 'is invalid';
+                break;
+            case 'password':
+                passwordValid = value.length >= 6;
+                fieldValidationErrors.password = passwordValid ? '': 'your password should contain at least 6 characters';
+                break;
+            case 'username':
+                usernameValid = value.length > 0;
+                fieldValidationErrors.username = usernameValid ? '': 'cannot be blank';
+                break;
+            case 'surname':
+                surnameValid = value.length > 0;
+                fieldValidationErrors.surname = surnameValid ? '': 'cannot be blank';
+                break;
+            case 'name':
+                nameValid = value.length > 0;
+                fieldValidationErrors.name = nameValid ? '': 'cannot be blank';
+                break;
+            default:
+                break;
+        }
+        this.setState({formErrors: fieldValidationErrors,
+            emailValid: emailValid,
+            passwordValid: passwordValid,
+            surnameValid: surnameValid,
+            nameValid: nameValid,
+            usernameValid: usernameValid
+        }, this.validateForm);
+    }
+
+    validateForm() {
+        if (this.state.signInView === false) {
```

Comment:
Shorten the boolean expression
```suggestion
        if (!this.state.signInView) {
```

(C'est top cette feature de GitHub de proposer des changements :D )